couchdb/server/__init__.py

```diff
@@ -0,0 +1,817 @@
+# -*- coding: utf-8 -*-
+#
+import logging
+import sys
+
+from couchdb import json, util
+from couchdb.server import (compiler, ddoc, exceptions, filters, render,
+                            state, stream, validate, views)
+from couchdb.server.helpers import partial, maybe_extract_source
+
+__all__ = ['BaseQueryServer', 'SimpleQueryServer']
+
+
+class NullHandler(logging.Handler):
+    def emit(self, *args, **kwargs):
+        pass
+
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.INFO)
+log.addHandler(NullHandler())
+
+
+class BaseQueryServer(object):
+    """Implements Python CouchDB query server.
+
+    :param version: CouchDB server version as three int elements tuple.
+                    By default tries to work against highest implemented one.
+    :type version: tuple
+
+    :param options: Custom keyword arguments.
+    """
+    def __init__(self, version=None, **options):
+        """Initialize query server instance."""
+
+        input_ = options.pop('input', sys.stdin)
+        output = options.pop('output', sys.stdout)
```

Comment:
How about making `input_` and `output` become parameters of `__init__` with default value?